Addon-docs: Packaging for 5.2 release (#7741)
storybookjs/storybook: addons/docs/docs/docspage.md

- [Motivation](#motivation)
- [Component parameter](#component-parameter)
- [DocsPage slots](#docspage-slots)
- [More resources](#more-resources)

## Motivation

`DocsPage` is the successor to [`addon-info`](https://github.com/storybookjs/storybook/tree/next/addons/info), which was one of the most popular Storybook addons despite many limitations.

Like `addon-info`, `DocsPage` provides sensible defaults, meaning it adds documentation to your existing Storybook without requiring any additional work on your part.

However, `DocsPage` brings the following improvements:

- It supports all frameworks that Storybook supports, including React, Vue, Angular and [many others](../README.md#framework-support).
- It generates better documentation that can be used as a standalone docs site, independently of Storybook.
- It supports better configuration, so you can capture project specific information with ease.
- It's built to work with [`MDX`](./mdx.md`) when you need more control of your documentation.

## Component parameter

> `DocsPage` pulls info from many sources, but one of the main ones is the `component` parameter, which is a new addition to Storybook in 5.2. It's based on the idea that each component should have an associated set of documentation and stories (versus organizing it in some other way).

Reviewer: "It's based on the idea" --> "It's based on best practices from the Storybook community-at-large"

> Storybook uses `component` to extract the component's description and props, and in future versions will probably do more with it. We encourage you to add it to existing stories and use it in all new stories.

Reviewer: ```suggestion
Storybook uses `component` to extract the component's description and props, and will rely on it further in future releases. We encourage you to add it to existing stories and use it in all new stories.
```

> Here's how to set the component in Component Story Format (CSF):

Reviewer: Link to CSF docs
